Makefile

```diff
@@ -1,73 +1,89 @@
-include .env.dev
-
-ENV_DIR ?= venv
-ENV = $(CURDIR)/$(ENV_DIR)
-CELERY = $(ENV)/bin/celery
-PRECOMMIT = $(ENV)/bin/pre-commit
-PIP = $(ENV)/bin/pip
-PYTHON3 = $(ENV)/bin/python3
-PYTEST = $(ENV)/bin/pytest
-
-DOCKER_FILE ?= docker-compose-developer.yml
-
-define setup_engine_env
-	export `grep -v '^#' .env | xargs -0` && cd engine
+DOCKER_COMPOSE_FILE = docker-compose-developer.yml
+
+# if .env is not present simply touch it
+$(shell touch .env)
+include .env
+
+# if COMPOSE_PROFILES is set in .env use it, otherwise use a default (or what is passed in as an arg)
+ifeq ($(COMPOSE_PROFILES),)
+	COMPOSE_PROFILES=engine,oncall_ui,redis,grafana
+endif
+
+# conditionally assign DB based on what is present in COMPOSE_PROFILES
+ifeq ($(findstring mysql,$(COMPOSE_PROFILES)),mysql)
+	DB=mysql
+else ifeq ($(findstring mysql,$(COMPOSE_PROFILES)),mysql)
+	DB=postgres
+else
+	DB=sqlite
+endif
+
+# conditionally assign BROKER_TYPE based on what is present in COMPOSE_PROFILES
+ifeq ($(findstring redis,$(COMPOSE_PROFILES)),redis)
+	BROKER_TYPE=redis
+else
+	BROKER_TYPE=rabbitmq
+endif
+
+define run_engine_docker_command
+    docker run -it \
```

Comment:
Probably it's better to use `docker-compose run` here? So there would be no need to specify the volume, env and network info.

Reply:
great suggestion 👍